src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -949,6 +949,81 @@ abstract class ManagedStrategy(
         }
     }
 
+    /**
+     * Tracks fields that are accessed via System.arraycopy, Unsafe API, VarHandle API, Java AFU API, and kotlinx.atomicfu.
+     */
+    private fun processMethodEffectOnStaticSnapshot(
+        owner: Any?,
+        className: String,
+        methodName: String,
+        params: Array<Any?>
+    ) = runInIgnoredSection {
+        // TODO:
+        //  1. tests for functionality below
+        //  2. for methods from `Arrays` class (sort, parallelSort, legacyMergeSort,
+        //     mergeSort, swap, parallelPrefix, fill, setAll, parallelSetAll).
+        //  3. for methods from `Collections` class (copyOf, copyOfRange)
+        when {
+            // System.arraycopy
+            className == "java/lang/System" && methodName == "arraycopy" -> {
+                //println("System.arraycopy: $className::$methodName")
+                check(params[2] != null && params[2] is Array<*>)
```

Comment:
`Array<*>` check will fail for primitive type arrays (like, `int[]`, `float[]`, `long[]`, ...).
Refactor this part